### PR TITLE
Add plugin: Friday

### PR DESCRIPTION
**Pull request by sunwei**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by sunwei** [comment]
> Thank you for your submission, an automated scan of your plugin code's revealed the following issues:
> 
> ### Required
> [[1](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/main.ts#L8-L10)][[2](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/main.ts#L109-L133)]:Remember to rename the sample classes to something that makes sense to your plugin. If there are any errors, Obsidian will spit out this class name, so this will be important for users to debug if they need to.
> 
> [[1](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L15)][[2](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L16)][[3](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L23)][[4](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L24)][[5](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L25)][[6](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L26)][[7](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L33)][[8](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/preview.ts#L34)]:You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.
> 
> [[1](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/fileinfo.ts#L48)][[2](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/fileinfo.ts#L103)][[3](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/main.ts#L48)][[4](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L42)][[5](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L53)][[6](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L193)][[7](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L222)][[8](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L255)][[9](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L279)][[10](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L305)][[11](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L328)][[12](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L415)][[13](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L435)][[14](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/hugoverse.ts#L443)]:You should consider limiting the number of `console.log`s in your code, to not pollute the dev console.
> 
> ### Optional
> [[1](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/server.ts#L49)][[2](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/main.ts#L85)]:Casting to `any` should be avoided as much as possible.
> 
> [[1](https://github.com/mdfriday/obsidian-friday-plugin/blob/572fa9b036fb6782c04c8fa197ca3376ddb1165e/src/main.ts#L103)]:Obsidian provides the [`requestUrl`](https://docs.obsidian.md/Reference/TypeScript+API/requestUrl) method, it will make sure that network requests work on every plattform.
> 
> Do NOT open a new PR for re-validation. Once you have pushed all of the required changes to your repo, the bot will update the labels on this PR within 6 hours. If you think some of the required changes are incorrect, please comment with `/skip` and the reason why you think the results are incorrect.

Updates according to all the suggestions above.

**Comment by sunwei** [comment]
Updated based on all the requests above.

**Comment by joethei** [comment]
[.my-modal {](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/styles.css#L2), [.my-iframe {](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/styles.css#L14), [modalEl.classList.add('my-modal');](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/preview.ts#L15), [iframe.classList.add('my-iframe');](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/preview.ts#L20)
Please change these selectors to something unique to your plugin.

["fundingUrl": "https://github.com/sunwei",](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/manifest.json#L9)
This is meant for links to services like _Buy me a coffee_, _GitHub sponsors_ and so on, if you don't have such a link remove this line.

[if (this.app.workspace.layoutReady) this.initLeaf()](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/main.ts#L43-L44)
This code should be simplified to
```ts
this.app.workspace.onLayoutReady(() => this.initLeaf())
```
If the layout is already ready then the code in the callback will be automatically executed.

[this.app.workspace.getLeavesOfType(FRIDAY_SERVER_VIEW_TYPE)[0]?.detach()](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/main.ts#L67)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[.setName('Netlify Token')](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/main.ts#L138)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[.setPlaceholder('Enter your netlify token')](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/main.ts#L141)
Netlify is a brand name, which means it's capitalization is defined by the owner of that brand, in this case it's capitalized.

[JSON.parse(response.text);](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/user.ts#L109)
Use `response.json` instead.

[this.name = `user_${Math.floor(Math.random() * 1000000)}@mdfriday.com`;](https://github.com/mdfriday/obsidian-friday-plugin/blob/e6996c7b496c88abdb990d3cf134399c3f531415/src/user.ts#L122-L123)

**Comment by sunwei** [comment]
Thank you for your valuable feedback! I have addressed all the requested changes and incorporated them into the latest release: [0.1.4](https://github.com/mdfriday/obsidian-friday-plugin/releases/tag/0.1.4).  

Please feel free to review the updates and let me know if there's anything else that needs attention. I truly appreciate your time and insights in helping improve the plugin!  

Thanks again! 😊

**Comment by joethei** [comment]
[.setName('Your netlify token')](https://github.com/mdfriday/obsidian-friday-plugin/blob/a1eb735822ed30bce3976f32a1c1f2e89a01da1c/src/main.ts#L213C19-L213C27)
Netlify is a brand name, which means it's capitalization is defined by the owner of that brand, in this case it's capitalized.

**Comment by waydesun** [comment]
Thank you for the prompt feedback! This is a straightforward fix, and I’ll address it shortly. The updated version will be submitted soon.

**Comment by sunwei** [comment]
Thank you for the prompt feedback! This is a straightforward fix, and I’ll address it shortly. The updated version will be submitted soon.

**Comment by sunwei** [comment]
Hi,  

Thank you for reviewing the Friday plugin. I have made the requested changes based on your feedback and have released a new version (**0.1.5**).  

The updated version is available here:  
[Release 0.1.5 on GitHub](https://github.com/mdfriday/obsidian-friday-plugin/releases/tag/0.1.5)  

Please let me know if there’s anything else that needs improvement.  

Thank you for your time and guidance!
